File: VeraPelle/src/main/java/psw/verapelle/DTO/ProductPublicDTO.java
// src/main/java/psw/verapelle/DTO/ProductPublicDTO.java
package psw.verapelle.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProductPublicDTO {
    private Long id;
    private String name;
    private BigDecimal price;
    private List<String> categoryNames;
    private String description;
    private List<String> imageUrls;
    private List<ColorDTO> colors;
    private Integer stockQuantity;
}

File: VeraPelle_front_standalone/src/app/core/models/product-public-dto.ts
// src/app/core/models/product-public-dto.ts

import { ColorDTO } from './color-dto';
import {ProductVariantDTO} from '@app/core/models/product-variant-dto';

export interface ProductPublicDTO {
  id: number;
  name: string;
  price: number;
  categoryNames: string[];
  description: string;
  imageUrls: string[];
  colors: ColorDTO[];             // puoi mantenerlo se serve altrove
  stockQuantity: number;         // totale (opzionale)
  variants: ProductVariantDTO[]; // ← qui prendi gli id/colori/hex/stock per il detail
}
